CoreScriptsRoot/LoadingScript.lua

```diff
@@ -762,3 +763,24 @@ game:GetService("ReplicatedFirst").RemoveDefaultLoadingGuiSignal:connect(handleR
 if game:GetService("ReplicatedFirst"):IsDefaultLoadingGuiRemoved() then
 	handleRemoveDefaultLoadingGui()
 end
+
+local UserInputServiceChangedConn;
+local function onUserInputServiceChanged(prop)
+	if prop == 'VREnabled' then
+		local UseVr = false
+		pcall(function() UseVr = UIS.VREnabled end)
+
+		if UseVr then
+			require(RobloxGui.Modules.LoadingScreen3D)
+			handleRemoveDefaultLoadingGui(true)
+			if UserInputServiceChangedConn then
```

Comment:
this seems like it would be better at the beginning of the if (its possible require could cause a yield). It shouldn't, but it could